### DISCUSSION TITLE
Help configuring firefly-iii self-hosted on apache2

### DISCUSSION THREAD

**Discussion by thiagotei** [firefly-iii/firefly-iii#8046] [Firefly III]
Looking for help configuring it on apache2.
I was successful installing firefly-iii standalone, but I am having problems when trying to setup a virtualhost so I can have multiple websites on the same server and access it by `locahost/firefly-iii`.
I can log in but clicking on any link results in error. For instance, clicking in reports goes to `localhost/reports` instead of `locahost/firefly-iii/reports`.
Do I need a rewrite rule to make it work?
Looking forward to use firefly!
Appreciate any help.

**Comment by JC5** [comment]
Virtualhosts aren't really meant to do localhost/whatever, but usually any-subdomain.domain.com. 

Firefly III and subdirectories isn't that complex though. What is your set up now?

**Reply by thiagotei** [reply]
Ubuntu 22.04, apache 2.4.

firefly-iii and data-importer are under `/var/www`.
I have the following in `/etc/apache2/sites-enabled/firefly-iii.conf` and I have disable the`000-default.conf`.

```
<VirtualHost *:80>
   Servername localhost
   DocumentRoot /var/www/html
   Alias /firefly-iii "/var/www/firefly-iii/public"
   Alias /data-importer "/var/www/data-importer"
</VirtualHost>
```
data-importer is working fine.
The firefly-iii works, but like I said, the links go to `localhost/<something>`. 
If I add `firefly-iii`  by hand, the `localhost/firefly-iii/<something>` works.

**Reply by JC5** [reply]
Does this help?

https://docs.firefly-iii.org/firefly-iii/faq/docker/#can-i-run-it-under-a-reverse-proxy-from-a-subdirectory

**Reply by thiagotei** [reply]
Yes! By using the `substitute` module I was able go around the issue. I added to the configuration above the following:
```
<Location "/firefly-iii">
AddOutputFilterByType SUBSTITUTE text/html
Substitute "s|href=\"/([\w-\d/]*)\"|href=\"firefly-iii/$1\"|i"
<\Location>
```
However, this seems hacky. I am not very knowledgeable on PHP/HTML/JS but seems that the base href is not being recognized:
```html
<base href="http://localhost/firefly-iii/)">
```
I could investigate further if you provide any leads.
Thanks for the help!

**Reply by thiagotei** [reply]
Not sure if it is related.
But I can't logout. I get a 405 error when I click to logout:
```
Oops! An Error Occurred
The server returned a "405 Method Not Allowed".
```

**Comment by JC5** [comment]
If you can, just use a subdomain and the docker container
